cpp/ql/lib/semmle/code/cpp/models/implementations/StdString.qll

```diff
@@ -15,15 +15,6 @@ private class StdBasicString extends ClassTemplateInstantiation {
   StdBasicString() { this.hasQualifiedName(["std", "bsl"], "basic_string") }
 }
 
-/**
- * The `std::basic_string::iterator` declaration.
- */
-private class StdBasicStringIterator extends Iterator, Type {
-  StdBasicStringIterator() {
-    this.getEnclosingElement() instanceof StdBasicString and this.hasName("iterator")
-  }
-}
```

Comment:
Are we completely confident one of the other `Iterator` extensions will catch this?

It does seem like a bit too much of a special case anyway.

Reply:
It certainly worked on https://github.com/github/codeql/blob/main/cpp/ql/test/library-tests/dataflow/taint-tests/stl.h. But I can run a DCA just to confirm.

Reply:
Ah, I see the warning about it above.  But `Iterator` itself is a public interface, so it could (in theory) affect someone's results.

Reply:
The class I deleted didn't actually contribute anything to the extent of the abstract class as it was already covered by this class: https://github.com/github/codeql/blob/ef72e222b0abef1c9b9a907011836d59e020a0d4/cpp/ql/lib/semmle/code/cpp/models/implementations/Iterator.qll#L64